src/middleware/parseUdi.ts

```diff
@@ -68,13 +65,22 @@ export const parseUdi = async (
         : "";
 
     // Split the pathSegment by periods to extract potential components
-    const parts = pathSegment.split(".");
+    const parts = pathSegment.split(":");
 
     if (parts.length === 1 && parts[0].length !== 64) {
       appendPath = `/${parts[0]}${appendPath}`;
     }
 
-    if (parts.length === 3) {
+    if (parts.length === 5) {

```

Comment:
It seems since we have the urn class, we might not need all this low level parsing login here?

Reply:
Agreed. I'll refactor that parsing into the udi class.